docs/pages/application-access/guides/usage.mdx

```diff
@@ -0,0 +1,273 @@
+  with DNS challenge, the below script will make setup easy. Update [EMAIL] and [DOMAIN]
+
+  ```sh
+  certbot certonly --manual \
+    --preferred-challenges=dns \
+    --email [EMAIL] \
+    --agree-tos \
+    --manual-public-ip-logging-ok \
+    -d "teleport.example.com, *.teleport.example.com"
+  ```
+</Admonition>
+
+### Create a User
+
+In order to be able to access the application, a Teleport user needs to have
+a role that allows them to see it.
+
+Teleport comes with a built-in `access` role that provides access to all
+applications:
+
+```sh
+$ tctl --config=/path/to/teleport.yaml users add --roles=access appuser
+```
+
+## Start Application Service with CLI Flags
+
+A Teleport Application Proxy agent can be started with a single CLI command:
+
+```sh
+$ teleport start \
+    --roles=app \
+    --token=/tmp/token \
+    --auth-server=teleport.example.com:3080 \
+    --app-name="grafana" \
+    --app-uri="http://localhost:3000" \
+    --labels=env=dev
+```
+
+Note that the `--auth-server` flag must point to the Teleport cluster's proxy
+endpoint because the application service always connects back to the cluster over
+a reverse tunnel.
+
+### Application Name
+
+When picking an application name, it's important to make sure the name will
+make a valid sub-domain (&lt;=63 characters, no spaces, only `a-z 0-9 _ -` allowed).
+
+After Teleport is running, the application will be accessible at
+`app-name.proxy_public_addr.com` e.g. `grafana.teleport.example.com`. Teleport
+also provides the ability to override `public_addr` e.g `grafana.acme.com`
+if you configure the appropriate DNS entry to point to the Teleport proxy server.
+
+## Start Application Service with Config File
+
+Example `/etc/teleport-app.yaml` configuration:
+
+```yaml
+teleport:
+  # Data directory for the Application Proxy service. If running on the same
+  # node as Auth/Proxy service, make sure to use different data directories.
+  data_dir: /var/lib/teleport-app
+  # Instructs the service to load the join token from the specified file
+  # during initial registration with the cluster.
+  auth_token: /tmp/token
+  # Proxy address to connect to. Note that it has to be the proxy address
+  # because the app service always connects to the cluster over a reverse
+  # tunnel.
+  auth_servers:
+  - teleport.example.com:3080
+app_service:
+    enabled: yes
+    # Teleport provides a small debug app that can be used to make sure application
+    # access is working correctly. It'll output JWTs so it can be useful
+    # when extending your application.
+    debug_app: true
+    # This section contains definitions of all applications proxied by this
+    # service. It can contain multiple items.
+    apps:
+      # Name of the application. Used for identification purposes.
+    - name: "grafana"
+      # URI and port the application is available at.
+      uri: "http://localhost:3000"
```

Comment:
This guide (or some other guide) should mention that if you have two apps with the same name on different hosts it will load balance them. (HA). May be for the future.

Reply:
Yeah I think dedicated HA guides for both app and db access are needed. Also, app access HA needs some work too, the way it's implemented right now it's not really HA because the session sticks to the app server for its entire duration.